src/gpu/amd/sycl_hip_engine.hpp

```diff
@@ -0,0 +1,99 @@
+/*******************************************************************************
+* Copyright 2020-2022 Intel Corporation
+*
+* Licensed under the Apache License, Version 2.0 (the "License");
+* you may not use this file except in compliance with the License.
+* You may obtain a copy of the License at
+*
+*     http://www.apache.org/licenses/LICENSE-2.0
+*
+* Unless required by applicable law or agreed to in writing, software
+* distributed under the License is distributed on an "AS IS" BASIS,
+* WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+* See the License for the specific language governing permissions and
+* limitations under the License.
+*******************************************************************************/
+
+#ifndef GPU_AMD_SYCL_HIP_ENGINE_HPP
+#define GPU_AMD_SYCL_HIP_ENGINE_HPP
+
+#include <stdexcept>
+#include "common/stream.hpp"
+#include "common/thread_local_storage.hpp"
+#include "gpu/amd/sycl_hip_utils.hpp"
+#include "miopen/miopen.h"
+#include "sycl/sycl_device_info.hpp"
+#include "sycl/sycl_engine_base.hpp"
+#include <CL/sycl.hpp>
+#include <hip/hip_runtime.h>
+
+namespace dnnl {
+namespace impl {
+namespace gpu {
+namespace amd {
+
+class hip_gpu_engine_impl_list_t {
+public:
+    static const impl_list_item_t *get_reorder_implementation_list(
+            const memory_desc_t *src_md, const memory_desc_t *dst_md);
+    static const dnnl::impl::impl_list_item_t *get_concat_implementation_list();
+    static const dnnl::impl::impl_list_item_t *get_sum_implementation_list();
+};
+
+class sycl_hip_engine_t : public dnnl::impl::sycl::sycl_engine_base_t {
+public:
+    using base_t = dnnl::impl::sycl::sycl_engine_base_t;
+
+    sycl_hip_engine_t(engine_kind_t kind, const ::sycl::device &dev,
+            const ::sycl::context &ctx, size_t index);
+    sycl_hip_engine_t(const ::sycl::device &dev, const ::sycl::context &ctx,
+            size_t index);
+
+    status_t create_stream(stream_t **stream, unsigned flags) override;
+    status_t create_stream(stream_t **stream, ::sycl::queue &queue);
+
+    const dnnl::impl::impl_list_item_t *get_reorder_implementation_list(
+            const memory_desc_t *src_md,
+            const memory_desc_t *dst_md) const override {
+        return nullptr;
```

Comment:
I'd suggest you to introduce an empty [cuda_reorder_impl_list](https://github.com/oneapi-src/oneDNN/blob/11fa74eaf03af9848c1bb5fffb4cbb2866aadf42/src/gpu/nvidia/cudnn_reorder_impl.cpp#L35) to return a pointer to it instead of `nullptr` because the code that calls `get_reorder_implementation_list` expects to get at least a pointer to `impl_list_item_t(nullptr)` otherwise there will be a segfault instead of returning `unimplemented` (e.g. [here](https://github.com/oneapi-src/oneDNN/blob/11fa74eaf03af9848c1bb5fffb4cbb2866aadf42/src/common/reorder.cpp#L91)).

The same applies to `get_concat_implementation_list` and `get_sum_implementation_list`.